app/models/user.rb

```diff
@@ -6,7 +6,68 @@ class User < ApplicationRecord
 
   validates :name, presence: true, length: { maximum: 20 }
 
+  # rubocop:disable Layout/LineLength:
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :recieved_friendship, class_name: 'Friendship', foreign_key: 'reciever_user_id', inverse_of: :reciever_user, dependent: :destroy
+  has_many :requested_friendship, class_name: 'Friendship', foreign_key: 'requester_user_id', inverse_of: :requester_user, dependent: :destroy
+  # rubocop:enable Layout/LineLength:
+  def friends
+    friends = []
+    Friendship.all.each do |request|
+      if request.status == true
+        if request.requester_user_id == id
+          friends << User.find(request.reciever_user_id)
+        elsif request.reciever_user_id == id
+          friends << User.find(request.requester_user_id)
+        end
+      end
+    end
+    friends
+  end
+
+  def pending_requests
+    pending_friend = []
+    Friendship.all.each do |pending|
+      next unless pending.status.nil?
+
+      pending_friend << pending if pending.reciever_user_id == id
+    end
+    pending_friend
+  end
```

Comment:
This case is the same as the above, instead, you will be looking for the status == nil.

I suggest that you also modify your DB so that every friendship you create has a default status of false (it's better than nil). Check [this article](https://stackoverflow.com/questions/7098602/add-a-default-value-to-a-column-through-a-migration) to know more.